### ISSUE TITLE
`promise-must-complete` false positive when passing resolve/reject out of scope

### ISSUE THREAD

**Issue by connor4312** [microsoft/tslint-microsoft-contrib#329]
Take the following code snippet that lazy-loads an external script:

```ts
new Promise(resolve => {
    const script = document.createElement('script');
    script.onload = resolve;
    script.src = 'http://example.com/script.js';
    document.body.appendChild(script);
});
```

This would fail with

```
demo.ts[1, 1]: A Promise was found that appears to not have resolve or reject invoked on all code paths
```

I would suggest providing a less-strict version of `promise-must-complete` rule, or an option,  which bails out and allows the rule to pass when `resolve` and/or `reject` are made available outside the scope of the promise's closure.

**Comment by HamletDRC** [comment]
Yes, so this was a conscious design decision. I like your recommendation of adding a parameter to make it a little less strict.

**Comment by JoshuaKGoldberg** [comment]
☠️  *It's time!* ☠️ 

Per #876, this repository is no longer accepting feature pull requests. TSLint is being deprecated and we recommend you switch to https://typescript-eslint.io.

Thanks for open sourcing with us, everyone!
